draft-ietf-ppm-dap.md

```diff
@@ -1498,30 +1531,57 @@ If the upload request is malformed, the Leader aborts with error
 If the Leader does not recognize the task ID, then it aborts with error
 `unrecognizedTask`.
 
+Otherwise, the Leader responds with a body consisting of an `UploadResponse` and
+the media type `application/dap-upload-resp`. The structure of the response
+is shown below:
+
+~~~ tls-presentation
+struct {
+  ReportID id;
+  ReportError error;
+} ReportUploadStatus;
+
+struct {
+  ReportUploadStatus status[status_count];
+} UploadResponse;
+~~~
+
+The Leader only includes reports that failed processing in the response.
+Reports that are accepted do not have a response.
+
+Reports in the response follow the same order as specified in the request.
+
+For each report that failed to upload, the Leader creates a
+`ReportUploadStatus` struct and includes the `ReportId` from the input and a
+`ReportError` {{basic-definitions}} that describes the failure. The length of
+this sequence is always less than or equal to the length of the upload sequence.
+
 If the Leader does not recognize the `config_id` in the encrypted input share,
-it aborts with an error of type `outdatedConfig`. When the Client receives an
-`outdatedConfig` error, it SHOULD invalidate any cached `HpkeConfigList` and
+it sets the error field to `outdated_config`
+(see {{aggregation-helper-init}}). When the Client receives an
+`outdated_config` error, it SHOULD invalidate any cached `HpkeConfigList` and
```

Comment:
```suggestion
it sets the error field to `outdated_config`. When the Client receives an
`outdated_config` error, it SHOULD invalidate any cached `HpkeConfigList` and
```
Is this reference correct? The `ReportError` enum was moved up in the document. Actually, since the preceding paragraph already points to {{basic-definitions}}, let's delete this altogether.